state/machine.go

```diff
@@ -2237,40 +2239,56 @@ func (m *Machine) SetUpgradeCharmProfile(appName, chURL string) error {
 				return nil, errors.Trace(err)
 			}
 		}
+
 		var emptyProfile bool
 		if ch == nil || (ch.LXDProfile() == nil || ch.LXDProfile().Empty()) {
 			emptyProfile = true
 		}
 
-		provisioned := true
+		ops := []txn.Op{
+			{
+				C:      machinesC,
+				Id:     m.doc.DocID,
+				Assert: isAliveDoc,
+			},
+			// TODO: lxd profile - enhance checks for this set of transactions
```

Comment:
this is my next card related to excessive content

Reply:
babbageclunk helped me find at least one excessive contention yesterday.  it's not new to this pr, but has been on going.  that work is being done in a different pr.  but perhaps there is more than one.

We're reading fresh, but also the charm itself was found from the same charmURL argument. It's a type issue of string vs charm.URL for the assert.